bluepill/src/BPPacker.m

```diff
@@ -97,6 +97,41 @@ @implementation BPPacker
     return bundles;
 }
 
++ (NSMutableArray<BPXCTestFile *> *)splitXCTestBundles:(NSArray<BPXCTestFile *> *)xcTestFiles
+                                   withTestsByFilePath:(NSMutableDictionary *)testsToRunByFilePath
+                           withTestEstimatesByFilePath:(NSMutableDictionary *)testEstimatesByFilePath
+                                         withTestTimes:(NSDictionary *)testTimes
+                                 withOptimalBundleTime:(double)optimalBundleTime {
+    NSMutableArray<BPXCTestFile *> *bundles = [[NSMutableArray alloc] init];
+    for (BPXCTestFile *xctFile in xcTestFiles) {
+        NSArray *bundleTestsToRun = [[testsToRunByFilePath[xctFile.testBundlePath] allObjects] sortedArrayUsingSelector:@selector(compare:)];
+        NSNumber *estimatedBundleTime = (NSNumber *)testEstimatesByFilePath[xctFile.testBundlePath];
```

Comment:
I think you only need the cast here because you're not passing value types above.